src/DeadCodeElimPass.cpp

```diff
@@ -63,15 +70,22 @@ bool DeadCodeElimPass::runOnFunction(llvm::Function &F) {
   size_t deleted = 0;
   UseSet use = computeUseSets(F);
 
+#if LLVM_VERSION_MAJOR >= 16
+  for (auto BBI = std::prev(F.end()); ; --BBI) {
+#else
   auto &BBL = F.getBasicBlockList();
   for (auto BBI = BBL.rbegin(); BBI != BBL.rend(); ++BBI) {
+#endif
```

Comment:
I guess I'd use `std::make_reverse_iterator` for this, something like [this](https://github.com/margnus1/nidhugg/commit/a815c55088daf67c40aa9cc8965a8cfee71d6158). Also, it seems like `llvm::Function` has begin() and end() already in llvm 8, so no `#if`-ing should be required.

Reply:
I just saw that this works. Do you want me to incorporate this change/commit as part of this PR, or you want to merge this fix with a separate PR to `master` first?

Reply:
I've incorporated this suggestion and squashed two commits of this PR.